encoding/m3tsz/iterator.go

```diff
@@ -35,33 +35,39 @@ import (
 // readerIterator provides an interface for clients to incrementally
 // read datapoints off of an encoded stream.
 type readerIterator struct {
-	is   *istream
-	opts Options
-	tess TimeEncodingSchemes
-	mes  MarkerEncodingScheme
+	is   encoding.Istream
+	opts encoding.Options
+	tess encoding.TimeEncodingSchemes
+	mes  encoding.MarkerEncodingScheme
 
 	// internal bookkeeping
 	t    time.Time     // current time
 	dt   time.Duration // current time delta
-	vb   uint64        // current value
-	xor  uint64        // current xor
+	vb   uint64        // current float value
+	xor  uint64        // current float xor
 	done bool          // has reached the end
 	err  error         // current error
 
+	intOptimized bool    // whether encoding scheme is optimized for ints
+	isFloat      bool    // whether encoding is in int or float
+	intVal       float64 // current int value
+	mult         uint8   // current int multiplier
+	sig          uint8   // current number of significant bits for int diff
+
 	ant       ts.Annotation // current annotation
 	tu        xtime.Unit    // current time unit
 	tuChanged bool          // whether we have a new time unit
-
-	closed bool
+	closed    bool
 }
 
 // NewReaderIterator returns a new iterator for a given reader
-func NewReaderIterator(reader io.Reader, opts Options) encoding.ReaderIterator {
+func NewReaderIterator(reader io.Reader, opts encoding.Options, intOptimized bool) encoding.ReaderIterator {
```

Comment:
Nit: can we make opts be the last parameter here? This is traditionally the pattern we have with other components that take opts as well as some parameters in their constructors.